### ISSUE TITLE
[BUG] Canceling expansion results in a volume expansion error

### ISSUE THREAD

**Issue by yangchiu** [longhorn/longhorn#9466]
## Describe the bug

Tested on v1.6.x-head (longhorn-manager 275f7ce). Expanding volume through UI and then Cancelling expansion through UI results in volume stuck in expansion error:

```
Expansion Error: the expected size 21474836480 of engine test-1-e-0 should not be smaller than the current size 22548578304
Note: You can cancel the expansion to avoid volume crash
```
![expand](https://github.com/user-attachments/assets/384e90cf-bbe6-4549-85bf-584bf1b8541a)

Try to cancel the expansion again, it shows another error:

```
unable to cancel expansion for volume test-1: volume expansion is not started
```

![expand6](https://github.com/user-attachments/assets/e7c46b2c-e6e6-473c-a1f0-6d29bb6ac327)

## To Reproduce

1. Create a 20 Gi volume through UI, and attach it to a node
![expand3](https://github.com/user-attachments/assets/35b4d2d8-43f7-43f0-8116-e015fbf183c5)

3. Expand the volume size to  21 Gi through UI
![expand5](https://github.com/user-attachments/assets/3075325c-43e8-4a7d-b36e-96a9d4b8b61e)

5. Before the expansion complete, cancel the expansion through UI
![expand2](https://github.com/user-attachments/assets/e19e377f-382b-4ccf-8cce-5e95c490c641)

7. Observe the volume gets stuck in Expansion error
![expand](https://github.com/user-attachments/assets/8fe743ac-8917-44ff-8b7b-aa8f614b0d30)

9. Try to cancel the expansion again, 
![expand6](https://github.com/user-attachments/assets/f3cb70d3-c109-414d-8972-197a8ef25182)

## Environment

 - Longhorn version: v1.6.x-head
 - Impacted volume (PV): 
 - Installation method (e.g. Rancher Catalog App/Helm/Kubectl): kubectl
 - Kubernetes distro (e.g. RKE/K3s/EKS/OpenShift) and version:
   - Number of control plane nodes in the cluster:
   - Number of worker nodes in the cluster:
 - Node config
   - OS type and version: ubuntu 24.04
   - Kernel version:
   - CPU per node:
   - Memory per node:
   - Disk type (e.g. SSD/NVMe/HDD):
   - Network bandwidth between the nodes (Gbps):
 - Underlying Infrastructure (e.g. on AWS/GCE, EKS/GKE, VMWare/KVM, Baremetal):
 - Number of Longhorn volumes in the cluster:

**Comment by c3y1huang** [comment]
@derekbit , I've added the auto-e2e-test label, and I encourage @longhorn/qa to submit a PR when they identify a bug. This will help developers reproduce the issue and verify the fix more efficiently.

**Comment by shuo-wu** [comment]
This issue seems to be triggered when the cancellation happens between the actual expansion completion and the engine/volume CR update. 
I don't think the [check](https://github.com/longhorn/longhorn-manager/blob/master/controller/volume_controller.go#L2029-L2030) @derekbit mentioned is the key. Actually, Longhorn cannot do anything with this case but just repeatedly reports that the cancellation request is invalid as the expansionist already done.

**Comment by shuo-wu** [comment]
Yes. Adding a warning log & a KB that asks the users to redo the expansion is all we can do.

**Comment by derekbit** [comment]
After rethinking this issue, the behavior is expected and we only need a KB for this issue. WDYT @shuo-wu @innobead

**Comment by shuo-wu** [comment]
I think updating the PR that includes the hint log is nice-to-have.

**Comment by davidcheng0922** [comment]
### RCA

Expansion cancellation is supposed to be allowed only before expansion starts, but the current check relies on `engine.Status.CurrentSize`, which can lag behind the actual backend state. 

This creates a race where backend expansion has already completed, but the engine CR still reports the old size, so `cancelExpansion` is incorrectly allowed and resets `volume.Spec.Size` to a smaller value. Reconciliation then hits the error: 
`engine ... should not be smaller than the current size ...`

Race Timeline

- T0: Expansion is requested.
- T1: engine.Spec.VolumeSize is updated, and backend expansion starts asynchronously.
- T2: A cancel expansion request arrives.
- T3: Backend expansion completes, but the result has not yet been reported back to the engine CR status.
- T4: engine.Status.CurrentSize still shows the old size, so cancellation is incorrectly allowed.
- T5: volume.Spec.Size is reverted to the original smaller size, even though the engine is already expanded.
- T6: Reconciliation sees a requested size smaller than the actual engine size, and the volume enters the error state.

### Fix Proposal

- Option 1: Completely disallow expansion cancellation.

- Option 2 (**current choice**): Check `engine.Spec.VolumeSize` instead of `engine.Status.CurrentSize`. Once `engine.Spec.VolumeSize` has been updated, treat the expansion request as already consumed by the controller and do not allow cancellation anymore.

- Option 3: Try to interrupt expansion in progress and shrink the volume back afterward. This is technically more complex and not worth the added risk or maintenance cost.

**Comment by davidcheng0922** [comment]
## Pre Ready-For-Testing Checklist
* [x] **Context & Reproduction**
    Where are the reproduction or test steps documented?
    Link:

1. Create a volume
2. Expand the volume
3. Directly cancel the expansion through UI
4. Regardless of whether the cancellation succeeds or fails, the volume must not become corrupted

* [ ] **Workaround**
    Is there a workaround for the issue? If yes, where is it documented?
    Link:

* [x] **Fix / Feature Explanation**
    Does the PR include a clear explanation of the fix or the new feature?

* [x] **Code Changes**
    Have all required code changes been merged?
    PR(s):
- https://github.com/longhorn/longhorn-manager/pull/5008

* [ ] **Deployment Changes (YAML / Helm Chart)**
    Does the PR introduce any deployment changes?
    PR:

* [ ] **Documentation Updates** *(if labeled `require/doc`)*
    Has the required documentation PR been submitted or merged?
    PR:

* [ ] **Manual Test Plan** *(if labeled `require/manual-test-plan`)*
    Has the manual test plan been documented?
    Link:
